airbyte-persistence/job-persistence/src/main/java/io/airbyte/persistence/job/JobPersistence.java

```diff
@@ -211,6 +211,10 @@ List<JobWithStatusAndTimestamp> listJobStatusAndTimestampWithConnection(UUID con
 
   Optional<Job> getLastSyncJob(UUID connectionId) throws IOException;
 
+  List<Optional<Job>> getLastSyncJobsForConnections(final List<UUID> connectionIds) throws IOException;
+
+  List<Optional<Job>> getRunningSyncJobForConnections(final List<UUID> connectionIds) throws IOException;
```

Comment:
Curious, why use a list of optionals here? If the list is empty, that suggest we don't have any items to return. Feels like we can simplify the type signature to a list of jobs. 

This would let us simplify the entire call chain -> JobHistoryHandler -> WebbackendConnectionsHandler etc.